File: src/features/cases/model/caseStatus.ts
import type { CaseStatus, EditableCaseStatus } from "./caseTypes";

export const editableCaseStatusOptions: ReadonlyArray<{
  value: EditableCaseStatus;
  label: string;
}> = [
  { value: "draft", label: "Черновик" },
  { value: "in_progress", label: "В работе" },
  { value: "prepared", label: "Подготовлено" },
  { value: "completed", label: "Завершено" },
];

export function isEditableCaseStatus(
  value: string
): value is EditableCaseStatus {
  return editableCaseStatusOptions.some((option) => option.value === value);
}

export function getCaseStatusLabel(status: CaseStatus): string {
  const option = editableCaseStatusOptions.find(
    (item) => item.value === status
  );

  if (option) {
    return option.label;
  }

  if (status === "archived") {
    return "Архив";
  }

  return status;
}

File: src-tauri/src/domain/case_status.rs
pub const CASE_STATUS_DRAFT: &str = "draft";
pub const CASE_STATUS_IN_PROGRESS: &str = "in_progress";
pub const CASE_STATUS_PREPARED: &str = "prepared";
pub const CASE_STATUS_COMPLETED: &str = "completed";
pub const CASE_STATUS_ARCHIVED: &str = "archived";

pub const EDITABLE_CASE_STATUSES: [&str; 4] = [
    CASE_STATUS_DRAFT,
    CASE_STATUS_IN_PROGRESS,
    CASE_STATUS_PREPARED,
    CASE_STATUS_COMPLETED,
];

pub fn is_editable_case_status(status: &str) -> bool {
    EDITABLE_CASE_STATUSES.contains(&status)
}
